unit_tests/test_lib_charms_ovn_charm.py

```diff
@@ -312,6 +322,21 @@ def test_optional_openstack_metadata(self):
         self.assertEquals(self.target.packages, ['ovn-host'])
         self.assertEquals(self.target.services, ['ovn-host'])
 
+    def test_is_port(self):
+        self.patch_object(
+            ovn_charm.subprocess, 'run',
+            side_effect=subprocess.CalledProcessError(1,
+                                                      'port does not exist'))
+        self.assertFalse(self.target.is_port('fake-port'))
+        self.run.assert_called_once_with(
+            ('ip', 'link', 'show', 'fake-port'),
+            stdout=ovn_charm.subprocess.PIPE,
+            stderr=ovn_charm.subprocess.STDOUT,
+            check=True,
+            universal_newlines=True)
+        self.patch_object(ovn_charm.subprocess, 'run', side_effect=None)
+        self.assertTrue(self.target.is_port('fake-port'))
```

Comment:
Maybe could use another self.run.assert_called_once_with() here? I think you'd need a reset_mock() before the 2nd is_port() call though.